### DISCUSSION TITLE
Does PMTiles compute tiles on the fly (only)? And what does it do for simplifying low zoom levels?

### DISCUSSION THREAD

**Discussion by deanm0000** [protomaps/PMTiles#488] [Q&A]
I recently went through an exercise of precomputing tiles for a map where I started at z=0 x=0 y=0 and then applied various levels of simplifying and merging of polygons as I iterated up to z=10+ with all the x/y along the way. Does PMTiles do this? In my version I can really notice from zoom level to zoom level a lack of smoothness as my simplification parameters change from zoom level to zoom level. When I zoom in/out in [https://pmtiles.io/?url=https%3A%2F%2Fr2-public.protomaps.com%2Fprotomaps-sample-datasets%2Fcb_2018_us_zcta510_500k.pmtiles#map=4.82/27.17/-82.65](this example), I don't notice anything like that.

I'm not sure if PMTiles is the tool to use for generating tiles in that way or if it's a server/file format and the simplifying as I described for low zoom levels should be another tool. If another tool, what is the other tool?

**Comment by bdon** [comment]
PMTiles is just the storage format, generally you need to use a (non-pmtiles specific) tile generator like https://github.com/felt/tippecanoe which does a good job at simplifying,

 If you need to generate tiles on-demand with a running tileserver process I would use pg_tileserv backed by a PostGIS instance.
